apps/state/lib/state/stops_on_route.ex

```diff
@@ -161,11 +165,15 @@ defmodule State.StopsOnRoute do
     |> Enum.uniq()
   end
 
-  defp order_stop_ids_for_trips(trips) do
-    trips
-    |> Stream.reject(&ignore_trip_for_route?/1)
-    |> stop_ids_for_trips
-    |> merge_ids
+  defp order_stop_ids_for_trips(route, direction_id, trips) do
+    trip_stops =
+      trips
+      |> Stream.reject(&ignore_trip_for_route?/1)
+      |> stop_ids_for_trips()
+
+    overrides = stop_order_overrides(route.id, direction_id)
```

Comment:
Could you reference `Application.get_env(:state, :stops_on_route)[:stop_order_overrides]` here directly?
It'd save some complexity with the macro definitions and the `@external_resource` above.